File: frontend/types/scheduleAgent.ts
export interface ScheduleRecommendation {
  recommended_date: string;
  product_name: string | null;
  recommended_quantity: number | null;
  unit: string | null;
  reasoning: string;
}

export interface ScheduleRecommendResponse {
  has_recommendation: boolean;
  recommendations: ScheduleRecommendation[];
  message: string;
}

File: backend/app/schemas/schedule_agent.py
from pydantic import BaseModel
from typing import Optional


class ScheduleRecommendRequest(BaseModel):
    year: int
    month: int


class ScheduleRecommendation(BaseModel):
    recommended_date: str
    product_name: Optional[str] = None
    recommended_quantity: Optional[int] = None
    unit: Optional[str] = None
    reasoning: str


class ScheduleRecommendResponse(BaseModel):
    has_recommendation: bool
    recommendations: list[ScheduleRecommendation]
    message: str
